Apeer/fiji_scripts/fijipytools.py

```diff
@@ -195,33 +195,22 @@ def readbf(imagefile, metainfo,
     @staticmethod
     def openjpg(imagefile,
                 method='IJ'):
-
-        if method == 'IJ':
-
-            # using IJ static method
-            imp = IJ.openImage(imagefile)
-
-        if method == 'Opener':
-
-            # Using Opener class
-            imp = Opener().openImage(imagefile)
-
-        if method == 'BF':
-
-            # using BioFormats library
-            imps = BF.openImagePlus(imagefile)
-
-            # read image data using the specified pyramid level
-            imp, slices, width, height, pylevel = ImageTools.getImageSeries(imps, series=readpylevel)
-
-            metainfo['Output Slices'] = slices
-            metainfo['Output SizeX'] = width
-            metainfo['Output SizeY'] = height
-
-            imp = imps[0]
-
-        return imp
-
+        allowed_methods = {"IJ": IJ.openImage, 
+                           "Opener": Opener().openImage, 
+                           "BF": BF.openImagePlus}
+        if method in allowed_methods:
```

Comment:
Good catch.

I am just wondering whether it is better to just omit the `if method in allowed_methods:` entirely here. It essentially boils down to the question whether `None` should be returned in this case or whether an exception should be raised. If an exception should be raised that you do not catch anyways, you can also omit the if clause.

Reply:
Committed with the omission as suggested